### DISCUSSION TITLE
Trying to add chezmoi.toml to chezmoi , but it fails

### DISCUSSION THREAD

**Discussion by Archie-2021** [twpayne/chezmoi#3285] [Q&A] [answered]
```
chezmoi add ~/.config/chezmoi/chezmoi.toml
chezmoi: ~/.config/chezmoi/chezmoi.toml: cannot add chezmoi file to chezmoi (~/.config/chezmoi/chezmoi.toml is protected)
```

Am I missing something here?

**Comment by twpayne** [comment]
Yes. See https://www.chezmoi.io/user-guide/setup/#create-a-config-file-on-a-new-machine-automatically.

**Reply by Archie-2021** [reply]
I am still confused after reading this, I don't want to initialize it. Should I only add chezmoi.toml with `--template`??

**Reply by halostatue** [reply]
The `chezmoi.toml` configuration file is explicitly *disallowed* from being added to the source, because it may have machine-specific configurations or even secrets.

When you create a `.chezmoi.toml.tmpl` or `.chezmoi.toml`, you can include anything you want, and then you do `chezmoi init --apply archie-2021/dotfiles` on a new machine and your dotfiles will be cloned, `~/.config/chezmoi/chezmoi.toml` will be created, and then everything will be applied (assuming you do not have other dependencies).

**Reply by twpayne** [reply]
Also, chezmoi reads its configuration file `chezmoi.toml` before making changes. So, if you try to manage `chezmoi.toml` then any changes you make will not take effect until the after the next time you run `chezmoi apply`. This is why the config file needs special treatment and is specifically protected.

**Reply by Archie-2021** [reply]
Thanks for your help. I see the reasoning behind it being this way. But is there a way to add it with my current settings? I only made cheMoi.toml to encrypt some files with my GPG key, I don't think that being a security issue for me as it just a reference to which key it should use.

P.S: Can I add it manually to the source directory? Or that would cause problems?

**Reply by bradenhilton** [reply, marked as answer]
@Archie-2021 You can probably just move your existing `chezmoi.toml` to your chezmoi source directory and rename it to `.chezmoi.toml.tmpl` then run `chezmoi init`.

By default, that would be `~/.local/share/chezmoi/.chezmoi.toml.tmpl`.

`chezmoi init` will execute your config template and store the result in `~/.config/chezmoi/chezmoi.toml`

**Reply by Archie-2021** [reply]
> @Archie-2021 You can probably just move your existing `chezmoi.toml` to your chezmoi source directory and rename it to `.chezmoi.toml.tmpl` then run `chezmoi init`.
> 
> By default, that would be `~/.local/share/chezmoi/.chezmoi.toml.tmpl`.
> 
> `chezmoi init` will execute your config template and store the result in `~/.config/chezmoi/chezmoi.toml`

Yeah is what I figured out after a bit of thinking and added to my post. So adding it manually to source is not going to be a problem.

Thanks for your help. Adding config of chezmoi to chezmoi recursively is very cool feature
